src/velocity_smoother.cpp

```diff
@@ -0,0 +1,413 @@
+/*
+ * Copyright (c) 2012, Yujin Robot.
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ *     * Redistributions of source code must retain the above copyright
+ *       notice, this list of conditions and the following disclaimer.
+ *     * Redistributions in binary form must reproduce the above copyright
+ *       notice, this list of conditions and the following disclaimer in the
+ *       documentation and/or other materials provided with the distribution.
+ *     * Neither the name of Yujin Robot nor the names of its
+ *       contributors may be used to endorse or promote products derived from
+ *       this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/*****************************************************************************
+ ** Includes
+ *****************************************************************************/
+
+#include <algorithm>
+#include <chrono>
+#include <cmath>
+#include <functional>
+#include <memory>
+#include <stdexcept>
+#include <string>
+#include <vector>
+
+#include <rcl_interfaces/msg/set_parameters_result.hpp>
+#include <rclcpp/rclcpp.hpp>
```

Comment:
Wondering if we're missing a dependency in `package.xml`. Getting a compile fail tracing back to here:

```
[ 25%] Building CXX object CMakeFiles/velocity_smoother.dir/src/velocity_smoother.cpp.o
/usr/bin/c++  -DPOCO_NO_AUTOMATIC_LIBS -DROS_PACKAGE_NAME=\"velocity_smoother\" -Dvelocity_smoother_EXPORTS -I/mnt/mervin/workspaces/eloquent/kobuki/src/velocity_smoother/include -I/opt/ros/eloquent/include  -fPIC   -Wall -Wextra -Werror -Wpedantic -std=c++14 -o CMakeFiles/velocity_smoother.dir/src/velocity_smoother.cpp.o -c /mnt/mervin/workspaces/eloquent/kobuki/src/velocity_smoother/src/velocity_smoother.cpp
In file included from /opt/ros/eloquent/include/rclcpp/subscription.hpp:44:0,
                 from /opt/ros/eloquent/include/rclcpp/any_executable.hpp:25,
                 from /opt/ros/eloquent/include/rclcpp/memory_strategy.hpp:24,
                 from /opt/ros/eloquent/include/rclcpp/memory_strategies.hpp:18,
                 from /opt/ros/eloquent/include/rclcpp/executor.hpp:33,
                 from /opt/ros/eloquent/include/rclcpp/executors/multi_threaded_executor.hpp:25,
                 from /opt/ros/eloquent/include/rclcpp/executors.hpp:21,
                 from /opt/ros/eloquent/include/rclcpp/rclcpp.hpp:145,
                 from /mnt/mervin/workspaces/eloquent/kobuki/src/velocity_smoother/src/velocity_smoother.cpp:44:
/opt/ros/eloquent/include/rclcpp/subscription_options.hpp: In member function ‘rcl_subscription_options_t rclcpp::SubscriptionOptionsWithAllocator<Allocator>::to_rcl_subscription_options(const rclcpp::QoS&) const’:
/opt/ros/eloquent/include/rclcpp/subscription_options.hpp:82:12: error: ‘rcl_subscription_options_t {aka struct rcl_subscription_options_t}’ has no member named ‘rmw_subscription_options’; did you mean ‘rcl_subscription_options_t’?
     result.rmw_subscription_options.ignore_local_publications = this->ignore_local_publications;
            ^~~~~~~~~~~~~~~~~~~~~~~~
            rcl_subscription_options_t
```

Reply:
Hm, that's weird.  I don't get anything similar here.  That actually kind of looks like either a mixed Dashing/Eloquent workspace, or some stuff from a source build of ROS 2 mixed in with Eloquent.  Can you try again from a completely clean terminal?  Otherwise we can sit down and look at this together.

Reply:
Looks like it may have been a deb that needed updating on my system. Did a full update and now it's compiling/launching fine.